subsys/bluetooth/host/hci_core.c

```diff
@@ -254,7 +254,10 @@ struct net_buf *bt_hci_cmd_create(uint16_t opcode, uint8_t param_len)
 	LOG_DBG("opcode 0x%04x param_len %u", opcode, param_len);
 
 	buf = net_buf_alloc(&hci_cmd_pool, K_FOREVER);
-	__ASSERT_NO_MSG(buf);
+	if (!buf) {
```

Comment:
Could you add a code comment here, explaining basically the same thing, since it's not obvious that unlike all other kernel objects that take timeout (which guarantee success with K_FOREVER) `net_buf_alloc()` has this special handling of the system work queue (which btw, I'm not convinced is not the best solution - feels more like a workaround for bad design elsewhere)

Reply:
I think we should rather do this special handling system-wide. Ie disallow `K_FOREVER` for any kernel objects when on the syswq. We can discuss this thursday.
I will also add the comment.